deploy-clojars/action.yml

```diff
@@ -0,0 +1,89 @@
+name: Clojars Deployment
+
+description: Build JAR and Deploy to Clojars
+
+inputs:
+  group-id:
+    description: The Clojars Group ID. Defaults to 'com.yetanalytics'.
+    required: false
+    default: 'com.yetanalytics'
+
+  artifact-id:
+    description: The Clojars Artifact ID (i.e. the name of the lib itself)
+    required: true
+
+  version:
+    description: >
+      The version string. Defaults to the Git branch or tag associated with
+      the event that triggered this action. NOTE: If the tag is prefixed
+      (e.g. 'v1.0.0'), you should remove the prefix, then pass in the
+      truncated tag as the arg (e.g. pass in `${GITHUB_REF#refs\/tags\/v}`).
+    required: false
+    default: ${{ github.ref_name }}
+
+  src-dirs:
+    description: >
+      A quoted array of the source directories. Defaults to '["src/main"]'.
+    required: false
+    default: '["src/main"]'
+
+  resource-dirs:
+    description: >
+      A quoted array of the resource directories. Defaults to '["resources"]'.
+      Pass '[]' if no resource directories exist.
+    required: false
+    default: '["resources"]'
+
+  # GitHub Secrets
+
+  clojars-username:
+    description: The Clojars username (should be stored as a GitHub secret).
+    required: true
+
+  clojars-password:
+    description: The Clojars password (should be an access token stored as a GitHub secret).
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Add aliases
+      shell: bash
+      run: >
+        clojure -Sdeps
+        '{:aliases
+          {:build
+           {:replace-deps {io.github.seancorfield/build-clj {:git/tag "v0.6.3" :git/sha "9b8e09b"}}
+            :ns-default build}}}'
+
+    - name: Build JAR file
+      shell: bash
+      run: >
+        clojure -X:build jar
+        :lib ${{ inputs.group-id }}/${{ inputs.artifact-id }}
+        :version '"${{ inputs.version }}"'
+        :src-dirs '${{ inputs.src-dirs }}'
+        :resource-dirs '${{ inputs.resource-dirs }}'
+        :scm '{
+          :url "https://github.com/${{ github.repository }}"
+          :tag "${{ github.sha }}"
+          :connection "scm:git:git://github.com/${{ github.repository }}.git"
+          :developerConnection "scm:git:ssh://git@github.com/${{ github.repository }}.git"
+        }'
+
+    # This is a debug step that exists for sanity checking
+    - name: Publish artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{ inputs.artifact-id }}-artifact-${{ inputs.version }}
+        path: target/${{ inputs.artifact-id }}-${{ inputs.version }}.jar
```

Comment:
If it is just for debug/sanity, perhaps lock down the retention to something short?

Reply:
Hmm maybe, but you're not going to be calling this action all that often (only on tag pushes), so I don't see the artifact retention length as a big deal here.

Reply:
true, and it's only gonna be an issue on private projects that won't be pushing to clojars anyways